src/watertap_contrib/reflo/core/solar_energy_base.py

```diff
@@ -301,7 +301,9 @@ def get_surrogate_data(
                 self.pickle_df = self.pickle_df[
                     (self.pickle_df[col] >= lo) & (self.pickle_df[col] <= hi)
                 ].copy()
-        self.data = self.pickle_df.sample(n=self.config.number_samples)
+        self.data = self.pickle_df.sample(
+            n=self.config.number_samples, random_state=len(self.pickle_df)
```

Comment:
What does including the ``random_state`` argument do here? Is this part of ensuring we are generating a consistent surrogate?

Reply:
Thanks for the quick review! My guess is this particular sample selection doesn't result in issues in the surrogate process.